ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ip-security.j2

```diff
@@ -0,0 +1,59 @@
+{#
+ Copyright (c) 2023 Arista Networks, Inc.
+ Use of this source code is governed by the Apache License 2.0
+ that can be found in the LICENSE file.
+#}
+{# eos - ip security #}
+{% if ip_security is arista.avd.defined %}
+!
+ip security
+{%     for ike_policy in ip_security.ike_policies | arista.avd.default([]) %}
+   !
+   ike policy {{ ike_policy.name }}
+{%         if ike_policy.local_id is arista.avd.defined %}
+      local_id {{ ike_policy.local_id }}
+{%         endif %}
+{%     endfor %}
+{%     for sa_policy in ip_security.sa_policies | arista.avd.default([]) %}
+   !
+   sa policy {{ sa_policy.name }}
+{%         if sa_policy.esp.intergrity is arista.avd.defined %}
+      esp intergrity {{ sa_policy.esp.intergrity }}
+{%         endif %}
+{%         if sa_policy.esp.encryption is arista.avd.defined %}
+      esp encryption {{ sa_policy.esp.encryption }}
+{%         endif %}
+{%         if sa_policy.pfs_dh_group is arista.avd.defined %}
+      pfs_dh_group {{ sa_policy.pfs_dh_group }}
+{%         endif %}
+{%     endfor %}
+{%     for profile in ip_security.profiles | arista.avd.default([]) %}
+   !
+   profile {{ profile.name }}
+{%         if profile.ike_policy is arista.avd.defined %}
+      ike-policy {{ profile.ike_policy }}
+{%         endif %}
+{%         if profile.sa_policy is arista.avd.defined %}
+      sa-policy {{ profile.sa_policy }}
+{%         endif %}
+{%         if profile.connection is arista.avd.defined %}
+      connection {{ profile.connection }}
+{%         endif %}
+{%         if profile.shared_key is arista.avd.defined %}
+      shared-key 7 {{ profile.shared_key }}
+{%         endif %}
+{%         if profile.dpd is arista.avd.defined %}
```

Comment:
missing variable checks for interval, time, action.

Reply:
they are required by schemas - do we still need to add the check in jinja?

```
              dpd:
                type: dict
                description: Dead Peer Detection.
                keys:
                  interval:
                    type: int
                    description: Interval (in seconds) between keep-alive messages.
                    required: true
                    convert_types:
                      - str
                    min: 2
                    max: 3600
                  time:
                    type: int
                    description: Time (in seconds) after which the action is applied.
                    required: true
                    convert_types:
                      - str
                    min: 10
                    max: 3600
                  action:
                    type: str
                    description: |
                      Action to apply

                      * 'clear': Delete all connections
                      * 'hold': Re-negotiate connection on demand
                      * 'restart': Restart connection immediately
                    required: true
                    valid_values:
                      - clear
                      - hold
  ```

Reply:
as discussed adding guards because people MAY not use the schema.. :)

Reply:
[9e3c696](https://github.com/aristanetworks/ansible-avd/pull/3194/commits/9e3c6961c943570d1a8ee0cfe6f739f9640f58f4)